src/vmod_soap.c

```diff
@@ -25,11 +25,14 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
+
+#include "cache/cache.h"
 #include "vmod_soap.h"
+#include "vcc_soap_if.h"
 
 #define POOL_KEY "VRN_IH_PK"
 
-static pthread_mutex_t	soap_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
what's the rationale behind this ?

Reply:
see https://github.com/Refinitiv/libvmod-soap/pull/8/commits/47ccd3ea4ee04ce1e2c5750f1b5d7bcdcee45bbf

> varnish-cache serializes the vmod events, so, unless the vmod itself has
> any concurent access, no mutex is required.